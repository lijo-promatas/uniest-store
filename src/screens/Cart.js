import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { View, Alert } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

// Import actions.
import * as cartActions from '../actions/cartActions';

// Components
import Spinner from '../components/Spinner';
import VendorsCartsList from '../components/VendorsCartsList';
import CartProductList from '../components/CartProductList';

// theme
import theme from '../config/theme';

// links
import { registerDrawerDeepLinks } from '../utils/deepLinks';
import i18n from '../utils/i18n';

import { iconsMap, iconsLoaded } from '../utils/navIcons';
import cart from '../reducers/cart';

// Styles
const styles = EStyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#FAFAFA',
	},
	topBtn: {
		padding: 10,
	},
	trashIcon: {
		height: 20,
		fontSize: 20,
	},
});

/**
 * Renders the cart modal.
 *
 * @reactProps {object} navigator - Navigator.
 * @reactProps {object} cartActions - Cart actions.
 * @reactProps {object} auth - Authorization information.
 * @reactProps {object} cart - Cart information.
 */
export class Cart extends Component {
	static navigatorStyle = {
		navBarBackgroundColor: theme.$navBarBackgroundColor,
		navBarButtonColor: theme.$navBarButtonColor,
		navBarButtonFontSize: theme.$navBarButtonFontSize,
		navBarTextColor: theme.$navBarTextColor,
		screenBackgroundColor: theme.$screenBackgroundColor,
	};

	/**
	 * @ignore
	 */
	static propTypes = {
		navigator: PropTypes.shape({
			push: PropTypes.func,
			dismissModal: PropTypes.func,
			setOnNavigatorEvent: PropTypes.func,
		}),
		cartActions: PropTypes.shape({
			fetch: PropTypes.func,
			clear: PropTypes.func,
			remove: PropTypes.func,
			change: PropTypes.func,
			changeAmount: PropTypes.func,
		}),
		auth: PropTypes.shape({
			token: PropTypes.string,
		}),
		cart: PropTypes.shape({}),
		vendorCarts: PropTypes.shape({}),
	};

	constructor(props) {
		super(props);

		this.state = {
			fetching: true,
			refreshing: false,
		};
		props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
	}

	/**
	 * Preloading icons. Sets title.
	 */
	componentWillMount() {
		const { navigator } = this.props;
		iconsLoaded.then(() => {
			navigator.setButtons({
				leftButtons: [
					{
						id: 'close',
						icon: iconsMap.close,
					},
				],
				rightButtons: [
					{
						id: 'clearCart',
						icon: iconsMap.delete,
					},
				],
			});
		});

		navigator.setTitle({
			title: i18n.t('Cart').toUpperCase(),
		});
	}

	/**
	 * Gets cart data.
	 */
	async componentDidMount() {
		const { cartActions } = this.props;
		await cartActions.fetch();
	}

	/**
	 * Updates the number of products in the state.
	 *
	 * @param {object} nextProps - Incoming props.
	 */
	componentWillReceiveProps(nextProps) {
		const { cart } = nextProps;

		if (cart.fetching) {
			return;
		}

		this.setState({
			fetching: false,
			refreshing: false,
		});
	}

	/**
	 * Cart modal navigation.
	 *
	 * @param {object} event - Information about the element on which the event occurred.
	 */
	onNavigatorEvent(event) {
		const { navigator, cartActions } = this.props;
		// handle a deep link
		registerDrawerDeepLinks(event, navigator);
		if (event.type === 'NavBarButtonPress') {
			if (event.id === 'close') {
				navigator.dismissModal();
			} else if (event.id === 'clearCart') {
				Alert.alert(
					i18n.t('Clear all cart ?'),
					'',
					[
						{
							text: i18n.t('Cancel'),
							onPress: () => {},
							style: 'cancel',
						},
						{
							text: i18n.t('Ok'),
							onPress: () => cartActions.clear(),
						},
					],
					{ cancelable: true }
				);
			}
		}
	}

	/**
	 * Refresh cart data.
	 */
	handleRefresh() {
		const { cartActions } = this.props;
		this.setState({ refreshing: true }, () => cartActions.fetch());
	}

	/**
	 * Renders a list of products.
	 *
	 * @return {JSX.Element}
	 */
	renderList() {
		const { refreshing, fetching } = this.state;
		const { cartActions, cart, auth, navigator } = this.props;

		if (fetching) {
			return this.renderSpinner();
		}

		return (
			<CartProductList
				cart={cart.carts.general}
				auth={auth}
				navigator={navigator}
				handleRefresh={this.handleRefresh}
				refreshing={refreshing}
				cartActions={cartActions}
			/>
		);
	}

	/**
	 * Renders a list of vendor carts.
	 *
	 * @return {JSX.Element}
	 */
	renderVendorsList = () => {
		const { fetching, refreshing } = this.state;
		const { cartActions, auth, navigator, cart } = this.props;

		if (fetching) {
			return this.renderSpinner();
		}

		const newCarts = Object.keys(cart.carts).reduce((result, el) => {
			if (el !== 'general') {
				result.push(cart.carts[el]);
			}
			return result;
		}, []);

		return (
			<VendorsCartsList
				carts={newCarts}
				auth={auth}
				navigator={navigator}
				handleRefresh={this.handleRefresh}
				refreshing={refreshing}
				cartActions={cartActions}
			/>
		);
	};

	/**
	 * Renders spinner.
	 *
	 * @return {JSX.Element}
	 */
	renderSpinner = () => {
		const { refreshing } = this.state;
		const { cart } = this.props;

		if (refreshing || !Object.keys(cart.carts).length) {
			return false;
		}

		return <Spinner visible={cart.fetching} />;
	};

	/**
	 * Renders component
	 *
	 * @return {JSX.Element}
	 */
	render() {
		const { cart } = this.props;
		return (
			<View style={styles.container}>
				{cart.isSeparateCart ? this.renderVendorsList() : this.renderList()}
			</View>
		);
	}
}

export default connect(
	(state) => ({
		auth: state.auth,
		cart: state.cart,
	}),
	(dispatch) => ({
		cartActions: bindActionCreators(cartActions, dispatch),
	})
)(Cart);
