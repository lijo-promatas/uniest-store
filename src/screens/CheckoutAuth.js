import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  View,
  ScrollView,
} from 'react-native';
import * as t from 'tcomb-form-native';
import EStyleSheet from 'react-native-extended-stylesheet';

// Import actions.
import * as authActions from '../actions/authActions';

// Components
import Spinner from '../components/Spinner';
import FormBlock from '../components/FormBlock';
import Button from '../components/Button';
import CheckoutSteps from '../components/CheckoutSteps';
import i18n from '../utils/i18n';

// theme
import theme from '../config/theme';

const styles = EStyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FAFAFA',
  },
  contentContainer: {
    padding: 14,
  },
});

const Form = t.form.Form;
const FormFields = t.struct({
  email: t.String,
  password: t.String,
});
const options = {
  disableOrder: true,
  fields: {
    email: {
      label: i18n.t('Email'),
      keyboardType: 'email-address',
      clearButtonMode: 'while-editing',
    },
    password: {
      label: i18n.t('Password'),
      secureTextEntry: true,
      clearButtonMode: 'while-editing',
    },
  }
};

class CheckoutAuth extends Component {
  static propTypes = {
    authActions: PropTypes.shape({
      login: PropTypes.func,
      logout: PropTypes.func,
    }),
    navigator: PropTypes.shape({
      setOnNavigatorEvent: PropTypes.func,
      setTitle: PropTypes.func,
      setStyle: PropTypes.func,
      dismissModal: PropTypes.func,
      showInAppNotification: PropTypes.func,
      push: PropTypes.func,
    }),
    auth: PropTypes.shape({
      logged: PropTypes.bool,
      error: PropTypes.string,
      fetching: PropTypes.bool,
    }),
  };

  static navigatorStyle = {
    navBarBackgroundColor: theme.$navBarBackgroundColor,
    navBarButtonColor: theme.$navBarButtonColor,
    navBarButtonFontSize: theme.$navBarButtonFontSize,
    navBarTextColor: theme.$navBarTextColor,
    screenBackgroundColor: theme.$screenBackgroundColor,
  };

  componentDidMount() {
    const { navigator } = this.props;
    navigator.setTitle({
      title: i18n.t('Login').toUpperCase(),
    });
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.auth.logged && !nextProps.auth.fetching) {
      setTimeout(() => {
        this.props.navigator.push({
          screen: 'CheckoutDelivery',
          backButtonTitle: '',
          passProps: {},
        });
      }, 1000);
    } else if (nextProps.auth.error && !nextProps.auth.fetching) {
      this.props.navigator.showInAppNotification({
        screen: 'Notification',
        passProps: {
          type: 'warning',
          title: i18n.t('Error'),
          text: i18n.t('Wrong password.')
        }
      });
    }
  }

  handleLogin() {
    const { authActions } = this.props;
    const value = this.refs.form.getValue();
    if (value) {
      authActions.login(value);
    }
  }

  handleLogout() {
    this.props.authActions.logout();
  }

  renderLoginForm() {
    return (
      <FormBlock title={i18n.t('Auth')}>
        <Form
          ref={'form'}
          type={FormFields}
          options={options}
        />
        <Button type="primary" onPress={() => this.handleLogin()}>
          {i18n.t('Sign in')}
        </Button>
      </FormBlock>
    );
  }

  renderReLogin() {
    return (
      <Button type="primary" onPress={() => this.handleLogout()}>
        {i18n.t('Sign in as a different user')}
      </Button>
    );
  }

  render() {
    const { auth } = this.props;
    return (
      <View style={styles.container}>
        <ScrollView
          contentContainerStyle={styles.contentContainer}
        >
          <CheckoutSteps step={0} />
          {auth.logged ? this.renderReLogin() : this.renderLoginForm()}
        </ScrollView>
        <Spinner visible={auth.fetching} mode="modal" />
      </View>
    );
  }
}

export default connect(state => ({
  auth: state.auth,
}),
dispatch => ({
  authActions: bindActionCreators(authActions, dispatch),
})
)(CheckoutAuth);
