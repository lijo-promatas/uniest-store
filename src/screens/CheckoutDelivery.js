import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { View } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

// Import components
import CheckoutSteps from '../components/CheckoutSteps';
import Spinner from '../components/Spinner';

// Import actions.
import * as authActions from '../actions/authActions';
import * as cartActions from '../actions/cartActions';

import i18n from '../utils/i18n';
import { formatPrice } from '../utils';

// theme
import theme from '../config/theme';
import ProfileForm from '../components/ProfileForm';

const styles = EStyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FAFAFA',
  },
  contentContainer: {
    paddingTop: 14,
    paddingBottom: 0,
    paddingLeft: 14,
    paddingRight: 14
  },
});

/**
 * Checkout. Delivery screen.
 *
 * @reactProps {object} navigator - Navigator.
 * @reactProps {object} cart - Cart information.
 * @reactProps {object} authActions - Auth actions.
 */
export class Checkout extends Component {
  static navigatorStyle = {
    navBarBackgroundColor: theme.$navBarBackgroundColor,
    navBarButtonColor: theme.$navBarButtonColor,
    navBarButtonFontSize: theme.$navBarButtonFontSize,
    navBarTextColor: theme.$navBarTextColor,
    screenBackgroundColor: theme.$screenBackgroundColor,
  };

  /**
   * @ignore
   */
  static propTypes = {
    navigator: PropTypes.shape({
      push: PropTypes.func,
      pop: PropTypes.func,
      setOnNavigatorEvent: PropTypes.func,
    }),
    cart: PropTypes.shape(),
    cartActions: PropTypes.shape(),
    authActions: PropTypes.shape(),
  };

  constructor(props) {
    super(props);
    this.state = {
      fieldsFetching: true
    };
    props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  /**
   * Gets fields and puts them in the state.
   */
  componentDidMount() {
    const { authActions } = this.props;
    const { fieldsFetching } = this.state;

    if (fieldsFetching) {
      authActions
        .profileFields({
          location: 'checkout',
          action: 'update'
        })
        .then(({ fields }) => {
          // eslint-disable-next-line no-param-reassign
          delete fields.E;

          this.setState({
            fields,
            fieldsFetching: false,
          });
        });
    }
  }

  /**
   * Delivery screen navigation.
   *
   * @param {object} event - Information about the element on which the event occurred.
   */
  onNavigatorEvent(event) {
    const { navigator } = this.props;
    if (event.type === 'NavBarButtonPress') {
      if (event.id === 'back') {
        navigator.pop();
      }
    }
  }

  /**
   * Saves user data, redirects to next screen.
   *
   * @param {object} values - Form data.
   */
  handleNextPress(values) {
    const { navigator, cart, cartActions } = this.props;

    cartActions.saveUserData({
      ...cart.user_data,
      ...values
    });

    navigator.push({
      screen: 'CheckoutShipping',
      backButtonTitle: '',
      title: i18n.t('Checkout').toUpperCase(),
      passProps: {
        total: cart.subtotal,
        cart
      },
    });
  }

  /**
   * Renders component
   *
   * @return {JSX.Element}
   */
  render() {
    const { cart } = this.props;
    const { fieldsFetching, fields } = this.state;

    if (fieldsFetching) {
      return (
        <View style={styles.container}>
          <Spinner visible />
        </View>
      );
    }

    return (
      <View style={styles.container}>
        <View style={styles.contentContainer}>
          <CheckoutSteps step={1} />
        </View>

        <ProfileForm
          fields={fields}
          cartFooterEnabled
          showTitles
          totalPrice={formatPrice(cart.total_formatted.price)}
          btnText={i18n.t('Next').toUpperCase()}
          onBtnPress={(values, validateCb) => { validateCb(); }}
          onSubmit={(values) => { this.handleNextPress(values); }}
        />
      </View>
    );
  }
}

export default connect(
  state => ({
    auth: state.auth,
    state,
  }),
  dispatch => ({
    authActions: bindActionCreators(authActions, dispatch),
    cartActions: bindActionCreators(cartActions, dispatch),
  })
)(Checkout);
