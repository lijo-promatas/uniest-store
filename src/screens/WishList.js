import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  View,
  Text,
  Image,
  Alert,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import Swipeout from 'react-native-swipeout';

// Import actions.
import * as wishListActions from '../actions/wishListActions';

// Components
import Icon from '../components/Icon';

// theme
import theme from '../config/theme';

// links
import { registerDrawerDeepLinks } from '../utils/deepLinks';
import i18n from '../utils/i18n';
import { formatPrice, getImagePath } from '../utils';

import {
  iconsMap,
  iconsLoaded,
} from '../utils/navIcons';

// Styles
const styles = EStyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FAFAFA',
  },
  productItem: {
    backgroundColor: '#fff',
    borderWidth: 1,
    borderColor: '#F1F1F1',
    flexDirection: 'row',
    paddingBottom: 10,
    padding: 14,
    width: '100%',
    overflow: 'hidden',
  },
  productItemImage: {
    width: 100,
    height: 100,
  },
  productItemDetail: {
    marginLeft: 14,
    width: '70%',
  },
  productItemWrapper: {
    marginBottom: 15,
  },
  productItemName: {
    fontSize: '0.9rem',
    color: 'black',
    marginBottom: 5,
    fontWeight: 'bold',
    textAlign: 'left'
  },
  productItemPrice: {
    fontSize: '0.7rem',
    color: 'black',
    textAlign: 'left'
  },
  emptyListContainer: {
    marginTop: '3rem',
    flexDirection: 'column',
    alignItems: 'center'
  },
  emptyListIconWrapper: {
    backgroundColor: '#3FC9F6',
    width: '12rem',
    height: '12rem',
    borderRadius: '6rem',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  emptyListIcon: {
    backgroundColor: 'transparent',
    color: '#fff',
    fontSize: '6rem',
  },
  emptyListHeader: {
    fontSize: '1.2rem',
    fontWeight: 'bold',
    color: 'black',
    marginTop: '1rem',
    paddingLeft: '0.25rem',
    paddingRight: '0.25rem',
    textAlign: 'center'
  },
  emptyListDesc: {
    fontSize: '1rem',
    color: '#24282b',
    marginTop: '0.5rem',
  },
});

export class WishList extends Component {
  static navigatorStyle = {
    navBarBackgroundColor: theme.$navBarBackgroundColor,
    navBarButtonColor: theme.$navBarButtonColor,
    navBarButtonFontSize: theme.$navBarButtonFontSize,
    navBarTextColor: theme.$navBarTextColor,
    screenBackgroundColor: theme.$screenBackgroundColor,
  };

  static propTypes = {
    navigator: PropTypes.shape({
      push: PropTypes.func,
      dismissModal: PropTypes.func,
      setOnNavigatorEvent: PropTypes.func,
    }),
    wishListActions: PropTypes.shape({
      fetch: PropTypes.func,
      remove: PropTypes.func,
      clear: PropTypes.func,
    }),
    wishList: PropTypes.shape({}),
  };

  constructor(props) {
    super(props);

    this.state = {
      fetching: true,
      refreshing: false,
    };

    props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  componentWillMount() {
    const { navigator, wishListActions } = this.props;

    wishListActions.fetch();
    iconsLoaded.then(() => {
      navigator.setButtons({
        leftButtons: [
          {
            id: 'close',
            icon: iconsMap.close,
          },
        ],
        rightButtons: [
          {
            id: 'clearWishList',
            icon: iconsMap.delete,
          },
        ],
      });
    });

    navigator.setTitle({
      title: i18n.t('Wish List').toUpperCase(),
    });
  }

  componentWillReceiveProps(nextProps) {
    const { wishList } = nextProps;
    if (wishList.fetching) {
      return;
    }

    this.setState({
      fetching: false,
      refreshing: false,
    });
  }

  onNavigatorEvent(event) {
    // handle a deep link
    registerDrawerDeepLinks(event, this.props.navigator);
    const { navigator } = this.props;
    if (event.type === 'NavBarButtonPress') {
      if (event.id === 'close') {
        navigator.dismissModal();
      } else if (event.id === 'clearWishList') {
        Alert.alert(
          i18n.t('Clear wish list?'),
          '',
          [
            {
              text: i18n.t('Cancel'),
              onPress: () => {},
              style: 'cancel'
            },
            {
              text: i18n.t('Ok'),
              onPress: () => this.props.wishListActions.clear(),
            },
          ],
          { cancelable: true }
        );
      }
    }
  }

  handleRemoveProduct = (product) => {
    const { wishListActions } = this.props;
    wishListActions.remove(product.cartId);
  };

  handleRefresh() {
    const { wishListActions } = this.props;
    this.setState(
      { refreshing: true },
      () => wishListActions.fetch(),
    );
  }

  renderProductItem = (item) => {
    let productImage = null;
    const imageUri = getImagePath(item);
    if (imageUri) {
      productImage = (
        <Image
          source={{ uri: imageUri }}
          style={styles.productItemImage}
        />);
    }

    const swipeoutBtns = [
      {
        text: i18n.t('Delete'),
        type: 'delete',
        onPress: () => this.handleRemoveProduct(item),
      },
    ];

    return (
      <View style={styles.productItemWrapper}>
        <Swipeout
          autoClose
          right={swipeoutBtns}
          backgroundColor={theme.$navBarBackgroundColor}
        >
          <TouchableOpacity
            style={styles.productItem}
            onPress={() => this.props.navigator.push({
              screen: 'ProductDetail',
              backButtonTitle: '',
              passProps: {
                pid: item.product_id,
                hideSearch: true,
                hideWishList: true,
              }
            })}
          >
            {productImage}
            <View style={styles.productItemDetail}>
              <Text
                style={styles.productItemName}
                numberOfLines={1}
              >
                {item.product}
              </Text>
              <Text style={styles.productItemPrice}>
                {item.amount} x {formatPrice(item.price_formatted.price)}
              </Text>
            </View>
          </TouchableOpacity>
        </Swipeout>
      </View>
    );
  }

  renderEmptyList = () => {
    if (this.state.fetching) {
      return null;
    }
    return (
      <View style={styles.emptyListContainer}>
        <View style={styles.emptyListIconWrapper}>
          <Icon name="favorite" style={styles.emptyListIcon} />
        </View>
        <Text style={styles.emptyListHeader}>
          {i18n.t('Your Wish Lists will love here.')}
        </Text>
      </View>
    );
  };

  renderList() {
    const { wishList } = this.props;
    return (
      <View style={styles.container}>
        <FlatList
          data={wishList.items}
          keyExtractor={(item, index) => `wishlist_${index}`}
          renderItem={({ item }) => this.renderProductItem(item)}
          onRefresh={() => this.handleRefresh()}
          refreshing={this.state.refreshing}
          ListEmptyComponent={() => this.renderEmptyList()}
        />
      </View>
    );
  }

  render() {
    return (
      <View style={styles.container}>
        {this.renderList()}
      </View>
    );
  }
}

export default connect(
  state => ({
    wishList: state.wishList,
  }),
  dispatch => ({
    wishListActions: bindActionCreators(wishListActions, dispatch),
  })
)(WishList);
