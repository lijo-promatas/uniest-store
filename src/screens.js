import { Provider } from 'react-redux';
import { Navigation } from 'react-native-navigation';

import Cart from './screens/Cart';
import Page from './screens/Page';
import Login from './screens/Login';
import Drawer from './screens/Drawer';
import Orders from './screens/Orders';
import Vendor from './screens/Vendor';
import Search from './screens/Search';
import Gallery from './screens/Gallery';
import Profile from './screens/Profile';
import Layouts from './screens/Layouts';
import CartBtn from './screens/CartBtn';
import WishList from './screens/WishList';
import Discussion from './screens/Discussion';
import Categories from './screens/Categories';
import WriteReview from './screens/WriteReview';
import OrderDetail from './screens/OrderDetail';
import ImagePicker from './screens/ImagePicker';
import VendorDetail from './screens/VendorDetail';
import CheckoutAuth from './screens/CheckoutAuth';
import Registration from './screens/Registration';
import ProductDetail from './screens/ProductDetail';
import Notification from './components/Notification';
import CheckoutPayment from './screens/CheckoutPayment';
import CheckoutComplete from './screens/CheckoutComplete';
import CheckoutShipping from './screens/CheckoutShipping';
import CheckoutDelivery from './screens/CheckoutDelivery';
import SettlementsCompleteWebView from './screens/SettlementsCompleteWebView';
import VendorManageOrders from './screens/VendorManage/Orders';
import VendorManageProducts from './screens/VendorManage/Products';
import VendorManageAddProductStep1 from './screens/VendorManage/AddProductStep1';
import VendorManageAddProductStep2 from './screens/VendorManage/AddProductStep2';
import VendorManageAddProductStep3 from './screens/VendorManage/AddProductStep3';
import VendorManageEditProduct from './screens/VendorManage/EditProduct';
import VendorManagePricingInventory from './screens/VendorManage/PricingInventory';
import VendorManageShippingProperties from './screens/VendorManage/ShippingProperties';
import VendorManageCategoriesPicker from './screens/VendorManage/CategoriesPicker';
import VendorManageOrderDetail from './screens/VendorManage/OrderDetail';

// register all screens of the app (including internal ones)
export default function registerScreens(store) {
  Navigation.registerComponent('Page', () => Page, store, Provider);
  Navigation.registerComponent('Cart', () => Cart, store, Provider);
  Navigation.registerComponent('Login', () => Login, store, Provider);
  Navigation.registerComponent('Drawer', () => Drawer, store, Provider);
  Navigation.registerComponent('Search', () => Search, store, Provider);
  Navigation.registerComponent('Orders', () => Orders, store, Provider);
  Navigation.registerComponent('Vendor', () => Vendor, store, Provider);
  Navigation.registerComponent('CartBtn', () => CartBtn, store, Provider);
  Navigation.registerComponent('Profile', () => Profile, store, Provider);
  Navigation.registerComponent('Gallery', () => Gallery, store, Provider);
  Navigation.registerComponent('Layouts', () => Layouts, store, Provider);
  Navigation.registerComponent('WishList', () => WishList, store, Provider);
  Navigation.registerComponent('Discussion', () => Discussion, store, Provider);
  Navigation.registerComponent('Categories', () => Categories, store, Provider);
  Navigation.registerComponent('WriteReview', () => WriteReview, store, Provider);
  Navigation.registerComponent('OrderDetail', () => OrderDetail, store, Provider);
  Navigation.registerComponent('VendorDetail', () => VendorDetail, store, Provider);
  Navigation.registerComponent('ImagePicker', () => ImagePicker, store, Provider);
  Navigation.registerComponent('Registration', () => Registration, store, Provider);
  Navigation.registerComponent('CheckoutAuth', () => CheckoutAuth, store, Provider);
  Navigation.registerComponent('Notification', () => Notification, store, Provider);
  Navigation.registerComponent('ProductDetail', () => ProductDetail, store, Provider);
  Navigation.registerComponent('CheckoutPayment', () => CheckoutPayment, store, Provider);
  Navigation.registerComponent('CheckoutDelivery', () => CheckoutDelivery, store, Provider);
  Navigation.registerComponent('CheckoutShipping', () => CheckoutShipping, store, Provider);
  Navigation.registerComponent('CheckoutComplete', () => CheckoutComplete, store, Provider);
  Navigation.registerComponent('SettlementsCompleteWebView', () => SettlementsCompleteWebView, store, Provider);
  Navigation.registerComponent('VendorManageOrders', () => VendorManageOrders, store, Provider);
  Navigation.registerComponent('VendorManageProducts', () => VendorManageProducts, store, Provider);
  Navigation.registerComponent('VendorManageAddProductStep1', () => VendorManageAddProductStep1, store, Provider);
  Navigation.registerComponent('VendorManageAddProductStep2', () => VendorManageAddProductStep2, store, Provider);
  Navigation.registerComponent('VendorManageAddProductStep3', () => VendorManageAddProductStep3, store, Provider);
  Navigation.registerComponent('VendorManageEditProduct', () => VendorManageEditProduct, store, Provider);
  Navigation.registerComponent('VendorManagePricingInventory', () => VendorManagePricingInventory, store, Provider);
  Navigation.registerComponent('VendorManageShippingProperties', () => VendorManageShippingProperties, store, Provider);
  Navigation.registerComponent('VendorManageCategoriesPicker', () => VendorManageCategoriesPicker, store, Provider);
  Navigation.registerComponent('VendorManageOrderDetail', () => VendorManageOrderDetail, store, Provider);
}
