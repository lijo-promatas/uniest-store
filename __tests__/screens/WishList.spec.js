import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { WishList } from '../../src/screens/WishList';

it('renders <WishList /> component correctly', () => {
  const tree = renderer.create(<WishList
    navigator={{
      pop: () => {},
      setTitle: () => {},
      dismissModal: () => {},
      showModal: () => {},
      setOnNavigatorEvent: () => {},
    }}
    wishListActions={{
      fetch: () => {},
    }}
    wishList={{
      items: [
        {
          main_pair: {
            detailed: {
              http_image_path: 'http://image.path'
            }
          },
          price_formatted: {
            price: '20$',
          }
        }
      ],
    }}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
