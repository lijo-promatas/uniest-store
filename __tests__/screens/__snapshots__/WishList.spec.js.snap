// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <WishList /> component correctly 1`] = `
<View>
  <View>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      data={
        Array [
          Object {
            "main_pair": Object {
              "detailed": Object {
                "http_image_path": "http://image.path",
              },
            },
            "price_formatted": Object {
              "price": "20$",
            },
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#dbddde",
                    "overflow": "hidden",
                  },
                  undefined,
                  Array [
                    Object {
                      "backgroundColor": "#FAFAFA",
                    },
                  ],
                ]
              }
            >
              <View
                onLayout={[Function]}
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Array [
                    Object {},
                    Object {
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                    />
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                       x 
                      20$
                    </Text>
                  </View>
                </View>
              </View>
              <View />
              <View />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
